src/Model/Resolver/PriceTiers.php

```diff
@@ -0,0 +1,230 @@
+            $this->customerGroupId = $this->getCustomerGroup->execute($context->getUserId());
+            $this->tiers = $this->tiersFactory->create(['customerGroupId' => $this->customerGroupId]);
+        }
+
+        $product = $value['model'];
+
+        if (
+            $product->hasData('can_show_price')
+            && $product->getData('can_show_price') === false
+            || !$product->getTierPrices()
+        ) {
+            return [];
+        }
+
+        $productId = (int)$product->getId();
+        $this->tiers->addProductFilter($productId);
+
+        return $this->valueFactory->create(
+            function () use ($productId, $context) {
+                // This lines added to clean previouse results
+                $this->formatAndFilterTierPrices = [];
+                $this->tierPricesQty = [];
+
+                $currencyCode = $context->getExtensionAttributes()->getStore()->getCurrentCurrencyCode();
+
+                $productPrice = $this->tiers->getProductRegularPrice($productId) ?? 0.0;
+                $tierPrices = $this->tiers->getProductTierPrices($productId) ?? [];
+
+                return $this->formatAndFilterTierPrices($productPrice, $tierPrices, $currencyCode);
+            }
+        );
+    }
+
+    /**
+     * Format and filter tier prices for output
+     *
+     * @param float $productPrice
+     * @param ProductTierPriceInterface[] $tierPrices
+     * @param string $currencyCode
+     * @return array
+     */
+    private function formatAndFilterTierPrices(
+        float $productPrice,
+        array $tierPrices,
+        string $currencyCode
+    ): array {
+
+        foreach ($tierPrices as $key => $tierPrice) {
+            $tierPrice->setValue($this->priceCurrency->convertAndRound($tierPrice->getValue()));
+            $this->formatTierPrices($productPrice, $currencyCode, $tierPrice);
+            $this->filterTierPrices($tierPrices, $key, $tierPrice);
+        }
+        return $this->formatAndFilterTierPrices;
+    }
+
+    /**
+     * Format tier prices for output
+     *
+     * @param float $productPrice
+     * @param string $currencyCode
+     * @param ProductTierPriceInterface $tierPrice
+     */
+    private function formatTierPrices(float $productPrice, string $currencyCode, $tierPrice)
+    {
+        $percentValue = $tierPrice->getExtensionAttributes()->getPercentageValue();
+        if ($percentValue && is_numeric($percentValue)) {
+            $discount = $this->discount->getDiscountByPercent($productPrice, (float)$percentValue);
+        } else {
+            $discount = $this->discount->getDiscountByDifference($productPrice, (float)$tierPrice->getValue());
+        }
+
+        $this->formatAndFilterTierPrices[] = [
+            "discount" => $discount,
+            "quantity" => $tierPrice->getQty(),
+            "final_price" => [
+                "value" => $tierPrice->getValue(),
+                "currency" => $currencyCode
+            ]
+        ];
```

Comment:
1. This variable is called the same as the function above
2. Please, if you collect one value from an array of other ones, use reduce instead of mutating some global state within this function
3. This function is called `formatSomething`, the name does not imply that it mutates the object it is called on. This comment should not cause any actions from your side, because comment 2 will make you rewrite this anyways.

Reply:
Hello!
I got the idea, but have question regarding those changes. Those functions are from extended resolver (parent), I added them into this one only because they have private access and didn't made any modifications in them. So before making changes in them question is: Do we need to change it and won't it be misleading in future?

Reply:
You currently have a function `formatAndFilterTierPrices` and an array `formatAndFilterTierPrices`, both private.
Being `private` implies not being inherited from anywhere, and not being inheritable from anywhere. Hence, you may call them as you wish. This is your implementation. 

Also, please make everything `protected` over `private` until it is __absolutely necessary__. We are extensibility-first organisation, and `private` stuff breaks the concept.